sim/core/statweight.go

```diff
@@ -173,8 +190,20 @@ func CalcStatWeight(swr proto.StatWeightsRequest, statsToWeigh []stats.Stat, ref
 
 	for _, stat := range statsToWeigh {
 		statMod := defaultStatMod
-		if stat == stats.SpellHit || stat == stats.MeleeHit || stat == stats.Expertise {
-			statMod = 15
+		if stat == stats.SpellHit {
+			statMod = spellHitStatMod
+			if baseStats[stat] < spellHitCap && baseStats[stat]+statMod > spellHitCap {
+				statModsHigh[stat] = baseStats[stat] - spellHitCap
```

Comment:
Instead of `baseStats[stat] - spellHitCap` why not just `-spellHitStatMod`? Example: if we're only 1 rating below cap, the current code here would give a stat mod of -1. That's way too low and will have huge amounts of noise.

Same comment for melee hit.

Reply:
Good point, the idea was simply to keep the code from going over the cap in its calculations, but that can indeed result in small mods that lead to high deviation. I'll think of something smarter